src/exchange/providers/ollama.py

```diff
@@ -0,0 +1,95 @@
+import os
+from typing import Any, Dict, List, Tuple, Type
+
+import httpx
+
+from exchange.message import Message
+from exchange.providers.base import Provider, Usage
+from exchange.providers.utils import (
+    messages_to_openai_spec,
+    openai_response_to_message,
+    openai_single_message_context_length_exceeded,
+    raise_for_status,
+    tools_to_openai_spec,
+)
+from exchange.tool import Tool
+
+OLLAMA_HOST = "http://localhost:11434/"
+
+
+
+#
+# NOTE: this is experimental, best used with 70B model or larger if you can. 
+# Example profile config to try:
+class OllamaProvider(Provider):
+    """Provides chat completions for models hosted by Ollama"""
+
+    """
+
+        ollama:
+          provider: ollama
+          processor: llama3.1
+          accelerator: llama3.1
+          moderator: passive
+          toolkits:
+          - name: developer
+            requires: {}
+    """    
+
+    def __init__(self, client: httpx.Client) -> None:
+        print('PLEASE NOTE: this is an experimental provider, use with care')
```

Comment:
maybe just mention here that OllamaProvider is the provider referenced (since this happens under the hood in goose, and might not be obvious)